### ISSUE TITLE
DFE doesn't support assert in initializers

### ISSUE THREAD

**Issue by aam** [dart-lang/sdk#29762]
On https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/painting/fractional_offset.dart#L50 DFE reports an error:

```
file:///flutter/flutter/packages/flutter/lib/src/painting/fractional_offset.dart:25:7: Error: Expected an expression, but got 'assert'.
    : assert(dx != null),
```

VM supports it as part of https://github.com/dart-lang/sdk/issues/27141.

**Comment by aam** [comment]
Hi folks, not sure if this is on your radar?

**Comment by peter-ahe-google** [comment]
This wasn't on my radar, and I don't recall seeing a language test that tests this.

**Comment by sigmundch** [comment]
This is covered by [assert_initializer_test.dart][1]

[1]: https://github.com/dart-lang/sdk/blob/463275881d3dfd9b1a33c3e0d51b74d8938c1188/tests/language/assert_initializer_test.dart

**Comment by peter-ahe-google** [comment]
Thank you, Siggi. This is extremely frustrating:

```
[$runtime != vm || $compiler != none]
assert_initializer_test: SKIP  # not implemented yet, experiment is VM only.
```

This just strengthens my resolve: we must get rid of Skip and SkipByDesign.

**Comment by aam** [comment]
@a-siva

**Comment by peter-ahe-google** [comment]
To be clear, it is now on my radar, and I plan to fix it ASAP, as I'm assuming you're blocked on this.

**Comment by aam** [comment]
Thanks Peter. I have removed all references to assert initializers from Flutter code for now, so no, we are not really blocked by this.

**Comment by peter-ahe-google** [comment]
Thank you, Alex. Then I'll be able to focus on metadata instead.

**Comment by aam** [comment]
Hi Peter, this will become a blocker soon. Is there anything I can do to help with this?

There are ~763 instances of asserts in initializers in Flutter code.

**Comment by peter-ahe-google** [comment]
Proposed fix in [CL 2930673002](https://codereview.chromium.org/2930673002/).

**Comment by aam** [comment]
Thank you!
